libs/ipfs-http-client/patches/001-add-srv-addr.patch

```diff
@@ -0,0 +1,247 @@
+@@ -24,9 +24,16 @@
+ #include <sstream>
+ #include <stdexcept>
+ 
+-int main(int, char**) {
++int main(int argc, char** argv) {
++  char addr127001[]="127.0.0.1";
++  char *srvaddr;
++  if (argc<2)
++    srvaddr=addr127001;
++  else
++    srvaddr=argv[1];
++
+   try {
+-    ipfs::Client client("localhost", 5001);
++    ipfs::Client client(srvaddr, 5001);
+ 
+     /** [ipfs::Client::BlockPut] */
+     ipfs::Json block;
+--- a/test/config.cc	2019-11-22 18:16:53.973272202 +0300
++++ b/test/config.cc	2019-11-22 18:16:58.245249328 +0300
+@@ -23,9 +23,16 @@
+ #include <iostream>
+ #include <stdexcept>
+ 
+-int main(int, char**) {
++int main(int argc, char** argv) {
++  char addr127001[]="127.0.0.1";
++  char *srvaddr;
++  if (argc<2)
++    srvaddr=addr127001;
++  else
++    srvaddr=argv[1];
++
+   try {
+-    ipfs::Client client("localhost", 5001);
++    ipfs::Client client(srvaddr, 5001);
+ 
+     /** [ipfs::Client::ConfigSet] */
+     client.ConfigSet("Datastore.StorageMax", "20GB");
+--- a/test/dht.cc	2019-11-22 18:16:53.973272202 +0300
++++ b/test/dht.cc	2019-11-22 18:16:58.245249328 +0300
+@@ -23,9 +23,16 @@
+ #include <iostream>
+ #include <stdexcept>
+ 
+-int main(int, char**) {
++int main(int argc, char** argv) {
++  char addr127001[]="127.0.0.1";
++  char *srvaddr;
++  if (argc<2)
++    srvaddr=addr127001;
++  else
++    srvaddr=argv[1];
++
+   try {
+-    ipfs::Client client("localhost", 5001);
++    ipfs::Client client(srvaddr, 5001);
+ 
+     ipfs::Json add_result;
+     client.FilesAdd(
+--- a/test/error.cc	2019-11-22 18:16:53.973272202 +0300
++++ b/test/error.cc	2019-11-22 18:16:58.245249328 +0300
+@@ -35,16 +35,23 @@
+ }
+ }  // namespace ipfs
+ 
+-int main(int, char**) {
++int main(int argc, char** argv) {
++  char addr127001[]="127.0.0.1";
++  char *srvaddr;
++  if (argc<2)
++    srvaddr=addr127001;
++  else
++    srvaddr=argv[1];
++
+   try {
+-    ipfs::Client client_cant_connect("localhost", 57);
++    ipfs::Client client_cant_connect(srvaddr, 57);
+ 
+     ipfs::test::must_fail("client.Version()", [&client_cant_connect]() {
+       ipfs::Json version;
+       client_cant_connect.Version(&version);
+     });
+ 
+-    ipfs::Client client("localhost", 5001);
++    ipfs::Client client(srvaddr, 5001);
+ 
+     std::string object_id;
+     client.ObjectNew(&object_id);
+--- a/test/files.cc	2019-11-22 18:16:53.973272202 +0300
++++ b/test/files.cc	2019-11-22 18:16:58.245249328 +0300
+@@ -24,9 +24,16 @@
+ #include <sstream>
+ #include <stdexcept>
+ 
+-int main(int, char**) {
++int main(int argc, char** argv) {
++  char addr127001[]="127.0.0.1";
++  char *srvaddr;
++  if (argc<2)
++    srvaddr=addr127001;
++  else
++    srvaddr=argv[1];
++
+   try {
+-    ipfs::Client client("localhost", 5001);
++    ipfs::Client client(srvaddr, 5001);
+ 
+     /** [ipfs::Client::FilesGet] */
+     std::stringstream contents;
+--- a/test/generic.cc	2019-11-22 18:16:53.973272202 +0300
++++ b/test/generic.cc	2019-11-22 18:16:58.245249328 +0300
```

Comment:
do you really need to patch all the tests? I guess it's the same as "are these tests useful for the openwrt build environment"

Reply:
these tests are useless on target device without this patch.